src/topology/sheaves/forget.lean

```diff
@@ -0,0 +1,209 @@
+begin
+  fsplit,
+  { intros S ι U,
+    -- We have that the sheaf condition fork for `F` is a limit fork,
+    have t₁ := S U,
+    -- and since `G` preserves limits, the image under `G` of this fork is a limit fork to.
+    have t₂ := @preserves_limit.preserves _ _ _ _ _ _ _ G _ _ t₁,
+    -- As we established above, that image is just the sheaf condition fork
+    -- for `F ⋙ G` postcomposed with some natural isomorphism,
+    have t₃ := is_limit.of_iso_limit t₂ (map_cone_fork G F U),
+    -- and as postcomposing by a natural isomorphism preserves limit cones,
+    have t₄ := is_limit.postcompose_inv_equiv _ _ t₃,
+    -- we have our desired conclusion.
+    exact t₄, },
+  { intros S ι U,
+    -- Let `f` be the universal morphism from `F.obj U` to the equalizer of the sheaf condition fork,
+    -- whatever it is. Our goal is to show that this is an isomorphism.
+    let f := equalizer.lift _ (w F U),
+    -- If we can do that,
+    suffices : is_iso (G.map f),
+    { resetI,
+      -- we have that `f` itself is an isomorphism, since `G` reflects isomorphisms
+      haveI : is_iso f := is_iso_of_reflects_iso f G,
+      -- TODO package this up as a result elsewhere:
+      apply is_limit.of_iso_limit (limit.is_limit _),
+      apply iso.symm,
+      fapply cones.ext,
+      exact (as_iso f),
+      rintro ⟨_|_⟩; { dsimp [f], simp, }, },
+    { -- Returning to the task of shwoing that `G.map f` is an isomorphism,
+      -- we note that `G.map f` is almost but not quite (see below) a morphism
+      -- from the sheaf condition cone for `F ⋙ G` to the
+      -- image under `G` of the equalizer cone for the sheaf condition diagram.
+      let c := sheaf_condition.fork (F ⋙ G) U,
+      have hc : is_limit c := S U,
+      let d := G.map_cone (equalizer.fork (left_res F U) (right_res F U)),
+      have hd : is_limit d := preserves_limit.preserves (limit.is_limit _),
+      -- Since both of these are limit cones
+      -- (`c` by our hypothesis `S`, and `d` because `G` preserves limits),
+      -- we hope to be able to conclude that `f` is a isomorphism.
+      -- We say "not quite" above because `c` and `d` don't quite have the same shape:
+      -- we need to postcompose by the natural isomorphism `diagram_comp_preserves_limits`
+      -- introduced above.
+      let d' := (cones.postcompose (diagram_comp_preserves_limits G F U).hom).obj d,
+      have hd' : is_limit d' :=
+        (is_limit.postcompose_hom_equiv (diagram_comp_preserves_limits G F U) d).symm hd,
+      -- Now everything works: we verify that `f` really is a morphism between these cones:
+      let f' : c ⟶ d' :=
+      fork.mk_hom (G.map f)
+      begin
+        dsimp only [c, d, d', f, diagram_comp_preserves_limits, res],
+        dunfold fork.ι,
+        ext1 j,
+        dsimp,
+        simp only [category.assoc],
+        simp only [←functor.map_comp_assoc],
+        simp only [equalizer.lift_ι, map_lift_comp_preserves_products_iso_hom_assoc],
```

Comment:
Maybe you split this up for a reason, but the proof goes through just as well if they are merged.

Reply:
```suggestion
        simp only [category.assoc, ←functor.map_comp_assoc, equalizer.lift_ι,
          map_lift_comp_preserves_products_iso_hom_assoc],
```